core/src/test/java/org/lflang/tests/compiler/LinguaFrancaValidationTest.java

```diff
@@ -172,6 +172,52 @@ public void disallowReactorCalledPreamble() throws Exception {
         "Reactor cannot be named 'Preamble'");
   }
 
+  @Test
+  public void requireSemicolonIfAmbiguous() throws Exception {
+    String testCase =
+        """
+            target C
+
+            reactor Foo {
+              output out: int
+              input inp: int
+              reaction(inp) -> out {==}
+            }
+
+            main reactor {
+              f1 = new Foo()
+              f2 = new Foo()
+              f3 = new Foo()
+
+              reaction increment(f1.out)
+              f2.out -> f3.inp
```

Comment:
Currently IIUC you emit an error and expect the reaction to be written
```

              reaction increment(f1.out)
              f2.out -> f3.inp;
```

which I believe doesn't help, as this still looks ambiguous.

I think the validator should not emit an error but a warning here, saying something like
```
              reaction increment(f1.out)
              f2.out -> f3.inp
              ^^^^^^^^^^^^ This line is parsed as a source and effects declaration for the reaction
                           on line 2, but it looks like a connection statement. If it is really a connection,
                           add a semicolon at the end of line 2. If it is really a dependency declaration, 
                           place it at the end of line 2 to silence this warning.
```

Reply:
My reasoning was that "at the end of the reaction declaration" would capture the intent because it would then be up to the programmer to specify where that end is (either at the error marker, or further down). I'm on board with making the message more explicit, like you suggest. Thanks for the thoughtful feedback, @oowekyala!

Reply:
You can also avoid emitting the message in certain cases by digging into the node model using NodeModelUtils.

The node model has all of the whitespace information that would be needed to make a very good guess about what the user really meant. For example, you can find out if there is a `\n` between `reaction increment(f1.out)` and `f2.out -> f3.inp`. You can also find out whether the line that says `f2.out -> f3.inp` is indented.

If we go this route, then the error message can recommend changing the whitespace instead of adding a semicolon.

Reply:
Handled this in 94e8da06f.